proto/auth.proto

```diff
@@ -69,8 +69,37 @@ message _GenerateApiTokenRequest {
         Expires expires = 2;
     }
 
-    // session token recieved from `momento login` command
-    string session_token = 3;
+    // Aliases for categories of functionality.
+    enum Role {
+      PermitNone = 0;
+      // Restricts access to apis that read and write data: No higher level resource description or modification.
+      ReadWrite = 1;
+    }
+
+    string auth_token = 3;
+
+    // Contains the permissions that the token will have for each of our services. If it doesn't exist
+    // on the request, then the token is assumed to have "SuperUser" permissions
+    message Permissions {
+      message CacheRestrictions {
+        Role role = 1;
+        // eventually could scope this down to a specific cache name or key
+        // string cache = "an example cache";
+        // string caches = ["cache1", "cache2"]
+      }
+
+      message TopicsRestrictions {
+        Role role = 1;
```

Comment:
A protobuf-type question: if this role enum is the same role enum for caches, can we later on break it apart into separate enums if we need to?

Like say later on I want to add a `foo` role for Caches but not for Topics - could we do that?

Reply:
no you would need to add another `TopicRole role2 = 2`, and deprecate the first role

Reply:
Yes - you can read the proto3 specification, but the short answer is this looks like `$tag_byte$enum_byte` on the wire. So it would be a series `12` for topicrestrictions readwrite. You could flip it to another enum, but that enum would need ordinals matching every previously possible value to be set. So we should never shoehorn in something that does not fit :-)

Reply:
matt's way is safer, and also documented in proto3 upgrade patterns

Reply:
ok I guess my question is should these be the same enum, or should we just have them be separate enums from the start?

Reply:
having them be separate enums would make it more flexible i think

Reply:
sure, is fine